app/models/mixins/compliance_mixin.rb

```diff
@@ -35,8 +37,16 @@ def scan_and_check_compliance_queue
   end
 
   def compliance_policies
-    target_class = self.class.base_model.name.downcase
-    _, plist = MiqPolicy.get_policies_for_target(self, "compliance", "#{target_class}_compliance_check")
+    _, plist = MiqPolicy.get_policies_for_target(self, "compliance", "#{event_towhat.downcase}_compliance_check")
     plist
   end
+
+  def event_towhat
```

Comment:
I think `event` is not granular enough since this is mixed in.  e.g. In vm.rb, what's this "event" referring to (could be an miq_event, a native event, a compliance event, etc).

```suggestion
  def compliance_policy_model_name
```

Reply:
Or even better, since you don't have overrides anymore, this can live directly in the `compliance_policies` method.

Reply:
Deleted

But the concept is real.

If you grep through the code for `class.base_model.name` and/or `match("template")` you will notice that this logic resides across a number of files, mostly centered around compliance, event processing, event names, and to some extent `towhat`. Tends to be called target_class.

you can find it in
- models/compliance.rb
- models/miq_policy.rb
- models/miq_alert.rb
- models/miq_event.rb

some of them are using underscore and others are using downcase (which has me concerned)
most of them remember to swap the name template over to vm